homeassistant/components/v2c/sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Support for V2C EVSE sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from pytrydan import TrydanData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import V2CUpdateCoordinator
+from .entity import V2CBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class V2CPowerRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[TrydanData], float]
+
+
+@dataclass
+class V2CPowerSensorEntityDescription(
+    SensorEntityDescription, V2CPowerRequiredKeysMixin
+):
+    """Describes an EVSE Power sensor entity."""
+
+
+POWER_SENSORS = (
+    V2CPowerSensorEntityDescription(
+        key="charge_power",
+        translation_key="charge_power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        value_fn=lambda evse_data: evse_data.charge_power,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up V2C sensor platform."""
+    coordinator: V2CUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[Entity] = [
+        V2CPowerSensorEntity(coordinator, description, config_entry.entry_id)
```

Comment:
We can use a generator expression directly in the call to `async_add_entities`.

Reply:
fixed